### PR TITLE
[BEAM-3655] Port MaxPerKeyExamplesTest off DoFnTester

### PR DESCRIPTION
**Pull request by niuk-a**
Hello @aromanenko-dev. I created solution for another ticket.

### CODE REVIEW

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java [hunk 1]
```diff
@@ -66,22 +71,27 @@
...
+  @Rule public TestPipeline p = TestPipeline.create();
+
   @Test
-  public void testExtractTempFn() throws Exception {
-    DoFnTester<TableRow, KV<Integer, Double>> extractTempFn = DoFnTester.of(new ExtractTempFn());
-    List<KV<Integer, Double>> results = extractTempFn.processBundle(TEST_ROWS);
-    Assert.assertThat(results, CoreMatchers.hasItem(kv1));
-    Assert.assertThat(results, CoreMatchers.hasItem(kv2));
-    Assert.assertThat(results, CoreMatchers.hasItem(kv3));
+  @Category(ValidatesRunner.class)
```
Comment (aromanenko-dev) [inline examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java]: I don't think we need to annotate these tests with `ValidatesRunner.class` category. 
@kennknowles could you confirm this?

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java [hunk 2]
```diff
@@ -66,22 +71,27 @@
...
+    p.run().waitUntilFinish();
   }
 
   @Test
-  public void testFormatMaxesFn() throws Exception {
-    DoFnTester<KV<Integer, Double>, TableRow> formatMaxesFnFn = DoFnTester.of(new FormatMaxesFn());
-    List<TableRow> results = formatMaxesFnFn.processBundle(TEST_KVS);
-    Assert.assertThat(results, CoreMatchers.hasItem(resultRow1));
-    Assert.assertThat(results, CoreMatchers.hasItem(resultRow2));
+  @Category(ValidatesRunner.class)
```
Comment (aromanenko-dev) [inline examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java]: The same question as above.

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java (hunk 1 above)
Comment (niuk-a) [reply examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java]: Hello Alexey. As I understand this annotation is used to check that test's author write pipeline.run
(p.run().waitUntilFinish();) in the end of the test. It has > 300 usages in all project. I think it is necessary here.

What should I do with this tests?
I can rewrite it like this:

`  @Rule public TestPipeline p = TestPipeline.create()`
`          .enableAbandonedNodeEnforcement(false)`
`         .enableAutoRunIfMissing(true);`
` @Test` public void testExtractTempFn() {`
`    PCollection<...> results = ...;`
`    PAssert.that(results).containsInAnyOrder(ImmutableList.of(kv1, kv2, kv3));`
`  }`

This approach will start pipeline automatically in the end of test (as I think)

Comment (aromanenko-dev) [reply examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java]: I think it's ok for now

**Pull request merged**